waku/v2/node/waku_node.nim

```diff
@@ -258,10 +263,32 @@ proc new*(T: type WakuNode,
     peerManager: PeerManager.new(switch, peerStorage),
     switch: switch,
     rng: rng,
-    enr: enr,
+    enr: populatedEnr,
     announcedAddresses: announcedAddresses
   )
 
+  if discv5Enabled:
+    var discv5Enrs: seq[enr.Record]
+    # parse enrURIs from the configuration and add the resulting ENRs to the discv5BootstrapEnrs seq
+    for enrUri in discv5BootstrapNodes:
+      addBootstrapNode(enrUri, discv5Enrs)
+    discv5Enrs.add(discv5BootstrapEnrs)
+
+    let rawMultiAddrField = enrMultiaddrs.getRawField()
+    # construct openArray[(string, seq[byte])]
+    let discv5EnrMultiAddrs = @[(MULTIADDR_ENR_FIELD, rawMultiAddrField)]
+    wakuNode.wakuDiscv5 = WakuDiscoveryV5.new(
```

Comment:
Perhaps change the `WakuDiscoveryV5` constructor to rather take a `seq[MultiAddress]` so the node does not need to know about enr construction?

Reply:
I think it is better set this way, since https://github.com/vacp2p/rfc/pull/566 refers to setting custom shard numbers. In a future PR, we can probably create an enum of `enrFieldPairs` which can be passed into the WakuDiscoveryV5 constructor, which all implement `getRawField` for use with the discv5 protocol.

```nim
type EnrFields = enum
  multiaddrs,
  shards
  
type EnrField* = object
  case key: EnrFields
  of multiaddrs:
    multiaddrs: seq[MultiAddress]
  of shards:
    shards: seq[ShardId]
  
getRawField(enrField: EnrField): seq[byte] =
  # same as getRawField implemented, but for ShardId and MultiAddress both
  if enrField.key == multiaddrs: # or a switch case
    return getRawField(enrField.multiaddrs)
  elif enrField.key == shards
    return getRawField(enrField.shards)

proc new*(T: type WakuDiscoveryV5,
          extIp: Option[ValidIpAddress],
          extTcpPort, extUdpPort: Option[Port],
          bindIP: ValidIpAddress,
          discv5UdpPort: Port,
          bootstrapNodes: seq[string],
          enrAutoUpdate = false,
          privateKey: keys.PrivateKey,
          flags: WakuEnrBitfield,
          enrFields: seq[EnrFields],
          rng: ref HmacDrbgContext,
          discv5Config: protocol.DiscoveryConfig = protocol.defaultDiscoveryConfig): T =
          var enrInitFields = @[(WAKU_ENR_FIELD, @[flags.byte])]
          for enrField in enrFields:
            let rawField = enrField.getRawField()
            enrInitFields.add((enrField.key, rawField))
          ...
```

Since the scaling solution will be put in place soon, I will not change the constructor to accept a seq[Multiaddress], wdyt?

Reply:
In that case I would either explicitly send the shard index and cluster number to the WakuDiscoveryV5 constructor or provide a field constructor elsewhere. The issue here is that we're mixing where the fields are initialised - `flags` for example is passed in as argument and added the enr fields internally, whereas the `multiaddrs` are not.

Reply:
Okay, I think I understand what you mean now. Addressed in [bffcca7](https://github.com/waku-org/nwaku/pull/1512/commits/bffcca70c8259ef80d2c88813d49074fef3197e6). I do think we should init the fields elsewhere in the future, though.